app/models/server/models/LivechatDepartmentAgents.js

```diff
@@ -65,10 +65,13 @@ export class LivechatDepartmentAgents extends Base {
 
 		const onlineUsernames = _.pluck(onlineUsers.fetch(), 'username');
 
+		// get the current agent load, sorted by the username & current chat count
+		const currentAvailableAgents = Promise.await(Users.getAgentLoadByUsernames(onlineUsernames, extraQuery));
```

Comment:
Why not just ignore the agents who are fully allocated?
As far as I understood, this PR intends to add a **restriction**, ignoring agents who can't be selected but the current implementation is doing the opposite, returning a list of agents who are supposed to be available for new chats.
So, in case there are 1k agents online a just 1 is fully allocated, the query will need to deal with 999 usernames rather than just one username. So, why not just ignore($nin) the agents who can't be selected?

Reply:
I think it's basically the same the opposite way. If you have 1 available agents and 999 fully allocated, the query will again deal with 999 users here. Which one is a more common scenario?

If a ton of available agents are more common than a ton of booked, I'll change the sense of the query, but we'll still have the issue you mention on "peak hours".

Reply:
The most common scenario is having just a few agents fully allocated, which means that this approach will usually return nothing and this is much better than always returning the list of all agents available, don't you think?

Reply:
Nice! I'll make the changes then.